### ISSUE TITLE
GET /api/task/info/<task_id> throws HTTP/1.1 500 Internal Server Error

### ISSUE THREAD

**Issue by sourav82** [mher/flower#617]
```
flower   -> flower:0.9.1 tornado:4.2 babel:2.3.4
software -> celery:4.0.0rc4 (0today8) kombu:4.0.0rc4 py:2.7.12
            billiard:3.5.0.1 py-amqp:2.1.0
platform -> system:Linux arch:64bit, ELF imp:CPython
loader   -> celery.loaders.default.Loader
settings -> transport:amqp results:disabled
```

API - GET /api/task/info/<task_id> throws below error always -

Flower server log -

```
[E 160925 10:44:06 web:1496] Uncaught exception GET /api/task/info/893dc5e8-1efe-4fb2-b648-5c64c72d8055 (::1)
    HTTPServerRequest(protocol='http', host='localhost:5555', method='GET', uri='/api/task/info/893dc5e8-1efe-4fb2-b648-5c64c72d8055', version='HTTP/1.1', remote_ip='::1', headers={'Host': 'localhost:5555', 'Accept': '*/*', 'User-Agent': 'curl/7.47.0'})
    Traceback (most recent call last):
      File "/usr/local/lib/python2.7/dist-packages/tornado/web.py", line 1413, in _execute
        result = method(*self.path_args, **self.path_kwargs)
      File "/usr/local/lib/python2.7/dist-packages/tornado/web.py", line 2717, in wrapper
        return method(self, *args, **kwargs)
      File "/usr/local/lib/python2.7/dist-packages/flower/api/tasks.py", line 602, in get
        self.write(response)
      File "/usr/local/lib/python2.7/dist-packages/tornado/web.py", line 697, in write
        chunk = escape.json_encode(chunk)
      File "/usr/local/lib/python2.7/dist-packages/tornado/escape.py", line 88, in json_encode
        return json.dumps(value).replace("</", "<\\/")
      File "/usr/lib/python2.7/json/__init__.py", line 244, in dumps
        return _default_encoder.encode(obj)
      File "/usr/lib/python2.7/json/encoder.py", line 207, in encode
        chunks = self.iterencode(o, _one_shot=True)
      File "/usr/lib/python2.7/json/encoder.py", line 270, in iterencode
        return _iterencode(o, 0)
      File "/usr/lib/python2.7/json/encoder.py", line 184, in default
        raise TypeError(repr(o) + " is not JSON serializable")
    TypeError: <_weakrefset.WeakSet object at 0x7f83e192b2d0> is not JSON serializable
[E 160925 10:44:06 web:1908] 500 GET /api/task/info/893dc5e8-1efe-4fb2-b648-5c64c72d8055 (::1) 7.06ms

```

Curl client log -

```
*   Trying ::1...
* Connected to localhost (::1) port 5555 (#0)
> GET /api/task/info/893dc5e8-1efe-4fb2-b648-5c64c72d8055 HTTP/1.1
> Host: localhost:5555
> User-Agent: curl/7.47.0
> Accept: */*
>
< HTTP/1.1 500 Internal Server Error
< Date: Sun, 25 Sep 2016 10:44:06 GMT
< Content-Length: 0
< Content-Type: text/html; charset=UTF-8
< Server: TornadoServer/4.2
<
* Connection #0 to host localhost left intact
```

Could you please let me know if I'm doing anything wrong here or any specific settings for this to work?

**Comment by tramora** [comment]
Hi all,

I encountered the same issue after the migration from celery 3.1.25 to 4.0.1 (and flower up to date 0.9.1). The fix #3490 on celery side is great and necessary but flower does not take benefit of it at as it makes it own serialization in the method TaskInfo::get

I suggest a pull request : https://github.com/mher/flower/pull/642/commits/1a93a9b8d5956da4137687ac1ed58ebaa15da269

**Comment by mher** [comment]
Fixed by https://github.com/mher/flower/commit/10cef872a7d4a4022dfdb02ff705b434f94103a7
